tests/lib/state_test.py

```diff
@@ -240,6 +240,39 @@ def test_absent_cannot_remove_absent(self):
         obj2 = state.RouteEntry(routes[0])
         assert not obj1.is_match(obj2)
 
+    def test_state_iface_routes(self):
+        routes = _get_mixed_test_routes()
+        route_state = state.State(
+            {
+                Route.KEY: {
+                    Route.CONFIG: routes
+                }
+            }
+        )
+
+        for route in routes:
+            assert (route in
+                    route_state.config_iface_routes.get(
+                        route[Route.NEXT_HOP_INTERFACE]))
+
+    def test_state_iface_routes_order(self):
+        # Changing all routes to eth1
```

Comment:
There is another test hiding here: When adding two routes with the same iface, they will both fall under the same iface index. Could you check this separately?